src/bodywork/git.py

```diff
@@ -44,102 +50,162 @@ def download_project_code_from_repo(
-               f'- currently, there is only support for HTTPS and SSL')
+        msg = (
+            f"cannot identify connection protocol in {connection_string}"
+            f"- currently, there is only support for HTTPS and SSH"
+        )
         raise RuntimeError(msg)
 
 
-def setup_ssh_for_github() -> None:
+def setup_ssh_for_git_host(hostname: str) -> None:
     """Setup system for SSH interaction with GitHub.
 
     Using the private key assigned to an environment variable, this
     function creates a new SSH configuration in the working directory
     and then tells Git to use it for SSH by exporting the
     GIT_SSH_COMMAND environment variable.
     """
-    ssh_dir = Path('.') / SSH_DIR_NAME
-    private_key = ssh_dir / 'id_rsa'
+    ssh_dir = Path(".") / SSH_DIR_NAME
+    private_key = ssh_dir / "id_rsa"
     if not private_key.exists():
-        if SSH_GITHUB_KEY_ENV_VAR not in os.environ:
-            msg = (f'failed to setup SSH for GitHub - cannot find '
-                   f'{SSH_GITHUB_KEY_ENV_VAR} environment variable')
+        if SSH_PRIVATE_KEY_ENV_VAR not in os.environ:
+            msg = (
+                f"failed to setup SSH for {hostname} - cannot find "
+                f"{SSH_PRIVATE_KEY_ENV_VAR} environment variable"
+            )
             raise KeyError(msg)
-        ssh_dir.mkdir(mode=0o700, exist_ok=True)
-        private_key.touch(0o700, exist_ok=False)
-        private_key.write_text(os.environ[SSH_GITHUB_KEY_ENV_VAR])
-
-    known_hosts = ssh_dir / 'known_hosts'
-    if not known_hosts.exists():
-        ssh_public_key = 'github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=='  # noqa
-        known_hosts.touch(0o700, exist_ok=False)
-        known_hosts.write_text(ssh_public_key)
-
-    os.environ['GIT_SSH_COMMAND'] = (
-        f'ssh -i {private_key} -o UserKnownHostsFile={known_hosts}'
+        try:
+            ssh_dir.mkdir(mode=0o700, exist_ok=True)
+            private_key.touch(0o700, exist_ok=False)
+            private_key.write_text(os.environ[SSH_PRIVATE_KEY_ENV_VAR])
+        except OSError as e:
+            raise RuntimeError(
+                f"Unable to create private key {private_key} from"
+                f" {SSH_PRIVATE_KEY_ENV_VAR} environment variable"
+            ) from e
+
+    try:
+        known_hosts = ssh_dir / "known_hosts"
+        if not known_hosts.exists():
+            known_hosts.touch(0o700, exist_ok=False)
+            known_hosts.write_text(get_ssh_public_key_from_domain(hostname))
+        elif not known_hosts_contains_domain_key(hostname, known_hosts):
+            known_hosts.touch(0o700)
```

Comment:
Why do you touch it again? You know it exists if you end-up here.

Reply:
That's a misunderstanding of the touch command on my part (it doesn't exist on Windows) I thought the mode was also looking at the file access rights but it doesn't.